src/main/java/com/google/sps/servlets/ProfileServlet.java

```diff
@@ -65,24 +64,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         boolean dairyFree = (boolean) entity.getProperty("dairyFree");
         ArrayList<String> allergies = (ArrayList<String>) entity.getProperty("allergies");
 
-        ProfileBuilder profileObjectBuilder = new ProfileBuilder(id, userName);
-        if (vegetarian) {
-          profileObjectBuilder.setVegetarian(vegetarian);
-        }
-        if (vegan) {
-          profileObjectBuilder.setVegan(vegan);
-        }
-        if (glutenFree) {
-          profileObjectBuilder.setGlutenFree(glutenFree);
-        }
-        if (dairyFree) {
-          profileObjectBuilder.setDairyFree(dairyFree);
-        }
-        if (allergies != null && allergies.size() > 0) {
-          profileObjectBuilder.setAllergies(allergies);
-        }
-
-        Profile profileObject = profileObjectBuilder.build();
+        Profile profileObject = Profile.builder().setId(id).setUserName(userName).
+          setVegetarian(vegetarian).setVegan(vegan).setGlutenFree(glutenFree).setDairyFree(dairyFree).
```

Comment:
How about if vegetarian/vega/... has a null value?

Reply:
Vegetarian/vegan/... are boolean values that cannot be null and are assigned the value false by default. Unless do you think it would be better to use Boolean values, that way they can be set to null?

Reply:
Since I changed the dietary needs properties to be a set of enums instead, the question about using Boolean no longer applies.